app/lib/task/backend.dart

```diff
@@ -0,0 +1,881 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:_pub_shared/data/task_api.dart' as api;
+import 'package:chunked_stream/chunked_stream.dart'
+    show readChunkedStream, MaximumSizeExceeded;
+import 'package:clock/clock.dart';
+import 'package:collection/collection.dart';
+import 'package:convert/convert.dart';
+import 'package:crypto/crypto.dart';
+import 'package:gcloud/service_scope.dart' as ss;
+import 'package:gcloud/storage.dart' show Bucket;
+import 'package:googleapis/storage/v1.dart'
+    show DetailedApiRequestError, ApiRequestError;
+import 'package:indexed_blob/indexed_blob.dart' show BlobIndex;
+import 'package:logging/logging.dart' show Logger;
+import 'package:pana/models.dart' show Summary;
+import 'package:pool/pool.dart' show Pool;
+import 'package:pub_dev/package/backend.dart' show packageBackend;
+import 'package:pub_dev/package/models.dart';
+import 'package:pub_dev/package/upload_signer_service.dart';
+import 'package:pub_dev/shared/datastore.dart';
+import 'package:pub_dev/shared/exceptions.dart';
+import 'package:pub_dev/shared/redis_cache.dart' show cache;
+import 'package:pub_dev/shared/utils.dart' show canonicalizeVersion;
+import 'package:pub_dev/shared/versions.dart'
+    show runtimeVersion, gcBeforeRuntimeVersion, shouldGCVersion;
+import 'package:pub_dev/task/cloudcompute/cloudcompute.dart';
+import 'package:pub_dev/task/global_lock.dart';
+import 'package:pub_dev/task/models.dart'
+    show PackageState, PackageVersionState, maxTaskExecutionTime;
+import 'package:pub_dev/task/scheduler.dart';
+import 'package:pub_worker/pana_report.dart' show PanaReport;
+import 'package:retry/retry.dart' show retry;
+import 'package:shelf/shelf.dart' as shelf;
+
+final _log = Logger('pub.task.backend');
+
+/// Register a [CloudCompute] pool for task workers in the current
+/// service scope.
+void registertaskWorkerCloudCompute(CloudCompute workerPool) =>
+    ss.register(#_taskWorkerCloudCompute, workerPool);
+
+/// Get the active [CloudCompute] pool for task workers.
+CloudCompute get taskWorkerCloudCompute =>
```

Comment:
Necessary for testing. Yes, I could omit it from the constructor.

I kind of wish we would migrate away from service scopes, so that's why I always end up with this thing that is half of each :see_no_evil: